src/main/java/org/apache/commons/collections4/bloomfilter/SetOperations.java

```diff
@@ -16,147 +16,172 @@
  */
 package org.apache.commons.collections4.bloomfilter;
 
-import org.apache.commons.collections4.bloomfilter.hasher.Shape;
+import java.util.function.LongBinaryOperator;
+import java.util.function.LongPredicate;
 
 /**
- * Implementations of set operations on Bloom filters.
+ * Implementations of set operations on BitMapProducers.
  *
+ * @since 4.5
  */
 public final class SetOperations {
```

Comment:
You have rewritten this class to entirely depend on the `makePredicate` idea where the predicate can send zeros if its array of bitmaps runs out before the other provider does. However this is not mirrored. A simple test with different shapes shows this fails:

```Java
        Shape shape = Shape.fromKM(3, 128);
        Shape shape2 = Shape.fromKM(3, 192);
        // ...
        filter1 = new SparseBloomFilter(shape2, IndexProducer.fromIntArray(new int[] { 1, 63, 185}));
        filter2 = new SparseBloomFilter(shape, IndexProducer.fromIntArray(new int[] { 5, 64, 69 }));
        // This fails!
        //assertEquals(6, SetOperations.orCardinality(filter1, filter2));
        assertEquals(6, SetOperations.orCardinality(filter2, filter1));
```